pennylane_lightning/lightning_qubit.py

```diff
@@ -528,7 +535,27 @@ def processing_fns(tapes):
 
         return processing_fns
 
-    def probability(self, wires=None, shot_range=None, bin_size=None):
+    def init_measures_object(self, pre_rotated=False):
+        # To support np.complex64 based on the type of self._state
+        dtype = self._state.dtype
+        if dtype == np.complex64:
+            use_csingle = True
+        elif dtype == np.complex128:
+            use_csingle = False
+        else:
+            raise TypeError(f"Unsupported complex Type: {dtype}")
+
+        # Initialization of state
+        state = self._pre_rotated_state if pre_rotated else self._state
+        ket = np.ravel(state)
+
+        if use_csingle:
+            ket = ket.astype(np.complex64)
+
+        state_vector = StateVectorC64(ket) if use_csingle else StateVectorC128(ket)
+        return MeasuresC64(state_vector) if use_csingle else MeasuresC128(state_vector)
+
+    def probability(self, wires=None, shot_range=None, bin_size=None, mc_obj=None):
```

Comment:
What is the `mc_obj` in this case? Should it be added to the docstring?

Since we are breaking from the super-class method signature, is this something we want to support externally, or internally only?

Do we expect this to go away with future implementations?

Lastly, would this be better as a class variable/entity and used throughout?